cmd/security-bootstrapper/entrypoint.sh

```diff
@@ -0,0 +1,233 @@
+# for the user may just want to debug into the container shell itself
+if [ "$1" = 'security-bootstrapper' ]; then
+    echo "Preparing ${EDGEX_VAULT_HOST} environment settings..."
+    cat >${SECURITY_INIT_DIR}/${VAULT_ENV_FILE} <<EOL
+BOOTSTRAPPER_HOST=${bootstrapper_host}
+WAIT_TCP_PORT=${bootstrap_port_number}
+WAIT_TIMEOUT_DURATION="${GATING_WAIT_TIMEOUT}"
+EOL
+
+    echo "Preparing ${vaultworker_host} environment settings..."
+    cat >${SECURITY_INIT_DIR}/${VAULT_WORKER_ENV_FILE} <<EOL
+VAULT_WORKER_HOST=${vaultworker_host}
+TOKENS_READY_PORT=${tokens_ready_port_number}
+EOL
+
+    echo "Preparing ${consul_host} environment settings..."
+    cat >${SECURITY_INIT_DIR}/${CONSUL_ENV_FILE} <<EOL
+CONSUL_HOST=${consul_host}
+CONSUL_PORT=${consul_port_number}
+CONSUL_READY_PORT=${consul_ready_port_number}
+EDGEX_VAULT_HOST="${EDGEX_VAULT_HOST}"
+EDGEX_VAULT_PORT=${EDGEX_VAULT_PORT}
+WAIT_TIMEOUT_DURATION="${GATING_WAIT_TIMEOUT}"
+EOL
+
+    echo "Preparing ${postgres_host} environment settings..."
+    cat >${SECURITY_INIT_DIR}/${POSTGRES_ENV_FILE} <<EOL
+BOOTSTRAPPER_HOST=${bootstrapper_host}
+WAIT_TCP_PORT=${bootstrap_port_number}
+EDGEX_VAULT_HOST="${EDGEX_VAULT_HOST}"
+EDGEX_VAULT_PORT=${EDGEX_VAULT_PORT}
+POSTGRES_HOST=${postgres_host}
+POSTGRES_PORT=${postgres_port_number}
+POSTGRES_READY_PORT=${postgres_ready_port_number}
+POSTGRES_PASSWORD_FILE="${POSTGRES_PASSWORD_FILE}"
+WAIT_TIMEOUT_DURATION="${GATING_WAIT_TIMEOUT}"
+EOL
+
+    echo "Preparing database ${redis_host} environment settings..."
+    cat >${SECURITY_INIT_DIR}/${REDIS_ENV_FILE} <<EOL
+VAULTWORKER_HOST=${vaultworker_host}
+WAIT_TCP_PORT=${tokens_ready_port_number}
+REDIS_HOST=${redis_host}
+REDIS_PORT=${redis_port_number}
+REDIS_READY_PORT=${redis_ready_port_number}
+WAIT_TIMEOUT_DURATION="${GATING_WAIT_TIMEOUT}"
+EOL
+
+    echo "Preparing kong environment settings..."
+    cat >${SECURITY_INIT_DIR}/${KONG_ENV_FILE} <<EOL
+BOOTSTRAPPER_HOST=${bootstrapper_host}
+WAIT_TCP_PORT=${ready_to_run_port_number}
+POSTGRES_HOST=${postgres_host}
+POSTGRES_PORT=${postgres_port_number}
+POSTGRES_READY_PORT=${postgres_ready_port_number}
+POSTGRES_PASSWORD_FILE="${POSTGRES_PASSWORD_FILE}"
+WAIT_TIMEOUT_DURATION="${GATING_WAIT_TIMEOUT}"
+EOL
+
+    echo "Preparing proxy-setup environment settings..."
+    cat >${SECURITY_INIT_DIR}/${PROXY_SETUP_ENV_FILE} <<EOL
+BOOTSTRAPPER_HOST=${bootstrapper_host}
+WAIT_TCP_PORT=${ready_to_run_port_number}
+PROXY_SETUP_HOST=${proxy_setup_host}
+KONG_HOST="${KONG_HOST}"
+KONG_STATUS_PORT=${KONG_STATUS_PORT}
+WAIT_TIMEOUT_DURATION="${GATING_WAIT_TIMEOUT}"
+EOL
+
+    echo "Preparing other services' environment settings..."
+    cat >${SECURITY_INIT_DIR}/${OTHERS_ENV_FILE} <<EOL
+BOOTSTRAPPER_HOST=${bootstrapper_host}
+WAIT_TCP_PORT=${ready_to_run_port_number}
+CONSUL_HOST=${consul_host}
+CONSUL_PORT=${consul_port_number}
+REDIS_HOST=${redis_host}
+REDIS_PORT=${redis_port_number}
+WAIT_TIMEOUT_DURATION="${GATING_WAIT_TIMEOUT}"
+EOL
```

Comment:
well, i have one issue with that: because this env value is generated dynamically and in one central place, so it is easier to maintain this way if wanted to be changed.   When all these envs are moved to compose files, then you'll have all duplicated values almost all over the place in compose-builder.  Let's discuss this more.

Reply:
ok, will work on this tomorrow.

Reply:
changed.